install/infra/single-cluster/gcp/README.md

```diff
@@ -0,0 +1,235 @@
+
+* `terraform` >= `v1.1.0`
+* `kubectl`   >= `v1.20.0`
+* [`jq`](https://stedolan.github.io/jq/download/)
+* [`gcloud`](https://cloud.google.com/sdk/docs/install)
+
+## Setup GCP authentication and GCS backend storage
+
+Before starting the installation process, you need:
+
+* A GCP account with administative access
+  * [Create one now by clicking here](https://console.cloud.google.com/freetrial)
+* Store the JSON credentials corresponding to the service account locally in a file
+* Create and configure GCS bucket for terraform backend
+  * Create a [GCS bucket](https://cloud.google.com/storage) to store the terraform backend state
+  * Replace the name of the bucket in [`main.tf`](./main.tf) - currently it is set as `gitpod-tf`
+
+## Update the `terraform.tfvars` file with appropriate values
+
+The file [`terraform.tfvars`](./terraform.tfvars) with values that will be used
+by terraform to create the cluster. While some of them are fairly
+straightforward like the name of the cluster(`cluster_name`), others need a bit
+more attention:
+
+### Credentials and project configuration
+
+To configure against a standing GCP account, we expect the the key corresponding
+to the service account stored as a JSON file. The path to the JSON file is
+expected to be provided as a value to the `credentials` field. Alongside, one is
+expected to provide the name of the project(`project` field) corresponding to
+this service account and region in with the cluster to be created(`region`
+field). If you want your cluster to be zonal(only existing in one zone), you can
+provide a zone corresponding to the project(`zone` field), else the cluster will
+be regional.
+
+### External database, storage and registry
+
+If you wish to create cloud specific database, storage and registry backend to be used
+with `Gitpod`, leave the following 3 booleans set:
+
+``` sh
+enable_external_database                     = true
+enable_external_storage                      = true
+enable_external_registry                     = true
+```
+
+The corresponding resources will be created by the terraform script which
+inclustes an `CloudSQL` mysql database, and access credentials to the GCS storage and GCR registry.
+
+The expectation is that you can use the credentials to these setups(provided later
+as terraform outputs) during the setup of Gitpod via UI later in the process.
+Alternatively, one can choose to use incluster dependencies or separately
+created resources of choice.
+
+### Kubernetes version
+
+Make sure you provide the corresponding kubernetes version as a value to the
+variable `cluster_version`. We officially support kubernetes versions >= `1.20`.
+
+### Domain name configuration
+
+If you are already sure of the domain name under which you want to setup Gitpod,
+we recommend highly to provide the value as `domain_name`. This will save a lot
+of hassle in setting up `Cloud DNS` records to point to the cluster and
+corresponding TLS certificate requests.
```

Comment:
**Preference:** I'd suggest wording this around what people might already have and actions they might need to take if they don't have our Terraform modules do the heavy lifting. Here's a quick pass at how that might be worded:

```suggestion
If you do not yet have a DNS zone created for Gitpod or plan on using cert-manager
to generate TLS certificates for gitpod, we strongly recommend setting `domain_name`
to a domain for use with Gitpod. This will save a lot
of hassle in setting up `Cloud DNS` records to point to the cluster and
corresponding TLS certificate requests.
```